daml-lf/scenario-interpreter/src/perf/benches/scala/com/digitalasset/daml/lf/speedy/perf/CollectAuthority.scala

```diff
@@ -15,27 +15,37 @@ import java.io.File
 import java.util.concurrent.TimeUnit
 import org.openjdk.jmh.annotations._
 
-@State(Scope.Thread)
+@State(Scope.Benchmark)
 class CollectAuthorityState {
-  private val darFile = new File(rlocation("daml-lf/scenario-interpreter/CollectAuthority.dar"))
-  private val packages = UniversalArchiveReader().readFile(darFile).get
-  private val packagesMap = Map(packages.all.map {
-    case (pkgId, pkgArchive) => Decode.readArchivePayloadAndVersion(pkgId, pkgArchive)._1
-  }: _*)
-
-  // NOTE(MH): We use a static seed to get reproducible runs.
-  private val seeding = crypto.Hash.secureRandom(crypto.Hash.hashPrivateKey("scenario-perf"))
-  private val buildMachine = Speedy.Machine
-    .newBuilder(
-      PureCompiledPackages(packagesMap).right.get,
-      Time.Timestamp.MinValue,
-      Some(seeding()))
-    .fold(err => sys.error(err.toString), identity)
-  private val expr = EVal(
-    Identifier(packages.main._1, QualifiedName.assertFromString("CollectAuthority:test")))
-  // NOTE(MH): We run the machine once to initialize all data that is shared
-  // between runs.
-  run()
+  private var buildMachine: Expr => Speedy.Machine = _
+  private var expr: Expr = _
+
+  @Param(Array("//daml-lf/scenario-interpreter/CollectAuthority.dar"))
+  private var dar: String = _
+  @Param(Array("CollectAuthority:test"))
+  private var scenario: String = _
+
+  @Setup(Level.Trial)
+  def init(): Unit = {
+    val darFile = new File(if (dar.startsWith("//")) rlocation(dar.substring(2)) else dar)
+    val packages = UniversalArchiveReader().readFile(darFile).get
+    val packagesMap = Map(packages.all.map {
+      case (pkgId, pkgArchive) => Decode.readArchivePayloadAndVersion(pkgId, pkgArchive)._1
+    }: _*)
```

Comment:
```suggestion
    val packagesMap = packages.all.map {
      case (pkgId, pkgArchive) => Decode.readArchivePayloadAndVersion(pkgId, pkgArchive)._1
    }.toMap
```

Reply:
I just copied that from `daml-lf/repl` and I have no clue what the weird smiley operator `: _*` exactly does. Your suggestion is definitely way easier to read.